src/components/quill-wrapper/quill-wrapper.vue

```diff
@@ -51,14 +51,13 @@ export default {
             this.quill.container.firstChild.innerHTML = this.value;
 
             this.quill.on("text-change", (delta, oldDelta, source) => {
-                if (source === "user") {
+                if (["user", "api"].includes(source)) {
                     const textHTML = this.quill.container.firstChild.innerHTML;
-
-                    if (!this.hasInnerText(textHTML)) {
-                        this.$emit("input", "");
+                    if (this.hasInnerText(textHTML) || this.hasInnerImage(textHTML)) {
```

Comment:
```suggestion
                    if (this.hasInnerText(textHTML) || textHTML.includes("<img")) {
```

Since Quill transforms any typed < or > to the HTML entities `&lt;` and `&gt;` you can safely ask specifically for the image opening tag.